pynautobot/core/endpoint.py

```diff
@@ -371,27 +405,36 @@ def __init__(self, parent_obj, name, custom_return=None):
         self.parent_obj = parent_obj
         self.custom_return = custom_return
         self.url = "{}/{}/{}/".format(parent_obj.endpoint.url, parent_obj.id, name)
-        self.request_kwargs = dict(base=self.url, token=parent_obj.api.token, http_session=parent_obj.api.http_session,)
 
-    def list(self, **kwargs):
+        self.request_kwargs = dict(
+            base=self.url,
+            token=parent_obj.api.token,
+            http_session=parent_obj.api.http_session,
+        )
+
+    def list(self, api_version=None, **kwargs):
         r"""The view operation for a detail endpoint
 
         Returns the response from Nautobot for a detail endpoint.
 
         Args:
+            api_version str, optional: Overide default or globally set Nautobot Rest API version for this(list) request.
             **kwargs: key/value pairs that get converted into url parameters when passed to the endpoint.
                 E.g. ``.list(method='get_facts')`` would be converted to ``.../?method=get_facts``.
 
         :returns: A dictionary or list of dictionaries retrieved from
             Nautobot.
         """
+        if api_version:
+            self.request_kwargs["api_version"] = api_version
+        
         req = Request(**self.request_kwargs).get(add_params=kwargs)
```

Comment:
Need to default to parent_obj.api.api_version, I think - also modifying self.request_kwargs seems wrong as that will persist to future calls on this same endpoint.

```suggestion
        api_version = api_version or self.parent_obj.api.api_version
        
        req = Request(api_version=api_version, **self.request_kwargs).get(add_params=kwargs)
```